packages/mars/src/execute/execute.ts

```diff
@@ -91,8 +108,13 @@ export async function getExistingDeployment(
   shouldSkipUpgrade: boolean,
   options: ExecuteOptions
 ): Promise<string | undefined> {
-  const existing = read(options.deploymentsFile, options.networkName, name)
-  if (existing) {
+  const existing = read<SaveEntry>(options.deploymentsFile, options.networkName, name)
+  if (!existing) return
+
+  if (existing.multisig) {
+    // TODO: multisig improvement candidate; for now we do not look for internal ex contract deployment data
+    return existing.address
```

Comment:
Hmm, it's not gonna be easy to check if contract has changed by tx.data. We should probably try comparing code under this address (provider.getCode(address) or something). This can be done later but it's quite important as contract redeployment wouldn't work now

Reply:
yes, the only option for now is manual deletion of deployment.json entry